aws/sqs_event_manager/sqs_event_manager/app.py

```diff
@@ -0,0 +1,133 @@
+    """
+    Expects event type matching the SNS topic
+
+    {
+        'Records': [
+            {
+                'message_id': 'str',
+                'receipt_handle': 'str',
+                'body': {
+                    'Type' : 'str',
+                    'MessageId : 'str',
+                    'TopicArn' : 'str',
+                    'Message' : {
+                        'action': 'str',
+                        'customer-identifier': 'str',
+                        'product-code': 'str',
+                    },
+                    'Timestamp' : 'str',
+                    'SignatureVersion' : 'str',
+                    'Signature' : 'str',
+                    'SigningCertURL' : 'str',
+                    'UnsubscribeURL' : 'str'
+                },
+                'attributes': {
+                    'ApproximateReceiveCount': 'str',
+                    'SentTimestamp': 'str',
+                    'SenderId': 'str',
+                    'ApproximateFirstReceiveTimestamp': 'str'
+                },
+                'messageAttributes': {},
+                'md5OfBody': 'str',
+                'eventSource': 'aws:sqs',
+                'eventSourceARN': 'arn:aws:sqs:us-east-1:111122223333:my-queue',
+                'awsRegion': 'us-east-1'
+            }
+        ]
+    }
+    """
+    batch_item_failures = []
+    sqs_batch_response = {}
+
+    try:
+        for message in event['Records']:
+            process_message(message, batch_item_failures)
+    except Exception as error:
+        return json.dumps(
+            error_response(500, f'{type(error).__name__}: {error}')
+        )
+
+    sqs_batch_response['batchItemFailures'] = batch_item_failures
+    return json.dumps(sqs_batch_response)
+
+
+def process_message(record: dict, batch_item_failures: dict):
+    """
+    Handle message received from SQS queue
+
+    Ensure the message is proper format. Get the customer entitlements and
+    send the information to the SCC service.
+    """
+    try:
+        message = AWSSNSMessage(record)
+
+        if not message.action:
+            logger.info(f'Received an unknown message: {json.dumps(record)}')
+        elif message.action == 'entitlement-updated':
+            # Notify SCC of customer entitlement change for the given product
+            entitlements = AWSCustomerEntitlement(
+                message.customer_id,
+                message.product_code
+            )
+            if entitlements.error:
+                return error_response(400, entitlements.error)
+
+            request = {  # noqa TODO: Cleanup noqa
+                'isBase64Encoded': False,
+                'statusCode': 200,
+                'body': {
+                    'CustomerIdentifier': message.customer_id,
+                    'ProductCode': message.product_code,
+                    'Entitlements': entitlements.get_entitlements()
+                }
+            }
+            # TODO: Send entitlement update request with entitlement info
```

Comment:
@schaefi @rjschwei do we know the details of this API yet?

Reply:
I don't know them. I wanted to actually see a real response send to the queue when a subscription changes